tutorials/plot_object_annotations.py

```diff
@@ -0,0 +1,172 @@
+# matches the recording time of the raw object.
+# Refer to the documentation of :class:`Annotations <mne.Annotations>` to see
+# the expected behavior depending on ``meas_date`` and ``orig_time``.
+# Where ``meas_date`` is the recording time of the stored in
+# :class:`Info <mne.Info>`.
+# You can find more information about :class:`Info <mne.Info>` in
+# :ref:`sphx_glr_auto_tutorials_plot_info.py`.
+#
+# We'll now manipulate some simulated annotations.
+# The first annotations has ``orig_time`` set to ``None`` while the
+# second is set to a chosen POSIX timestamp for illustration purposes.
+
+###############################################################################
+
+# Create an annotation object without orig_time
+annot_none = mne.Annotations(onset=[0, 2, 9], duration=[0.5, 4, 0],
+                             description=['foo', 'bar', 'foo'],
+                             orig_time=None)
+print(annot_none)
+
+# Create an annotation object with orig_time
+orig_time = '2002-12-03 19:01:31.676071'
+my_datetime = datetime.strptime(orig_time, '%Y-%m-%d %H:%M:%S.%f')
+posix_timestamp = (my_datetime - datetime(1970, 1, 1)).total_seconds()
+print('{0:f} is the POSIX timestamp of {1:s}'.format(posix_timestamp,
+                                                     orig_time))
+
+annot_orig = mne.Annotations(onset=[22, 24, 31], duration=[0.5, 4, 0],
+                             description=['foo', 'bar', 'foo'],
+                             orig_time=1038942091.6760709)
+print(annot_orig)
+
+###############################################################################
+# Now we create two raw objects, set the annotations and plot them to compare
+# them.
+
+# Create two cropped copies of raw with the two previous annotations
+raw_a = raw.copy().crop(tmax=12).set_annotations(annot_none)
+raw_b = raw.copy().crop(tmax=12).set_annotations(annot_orig)
+
+# Plot the raw objects
+raw_a.plot()
+raw_b.plot()
+
+# Show the annotations in the raw objects
+print(raw_a.annotations)
+print(raw_b.annotations)
+
+# Show that the onsets are the same
+print(raw_a.annotations.onset)
+print(raw_b.annotations.onset)
+
+###############################################################################
+#
+# Notice that for the case where ``orig_time`` is ``None``,
+# one assumes that the orig_time is the time of the first sample of data.
+
+raw_delta = (1 / raw.info['sfreq'])
+print('raw.first_sample is {}'.format(raw.first_samp * raw_delta))
+print('annot_none.onset[0] is {}'.format(annot_none.onset[0]))
+print('raw_a.annotations.onset[0] is {}'.format(raw_a.annotations.onset[0]))
+
+###############################################################################
+#
+# It is possible to concatenate two annotations with the + operator like for
+# lists if both share the same ``orig_time``
+
+annot = mne.Annotations(onset=[10], duration=[0.5],
+                        description=['foobar'],
+                        orig_time=1038942091.6760709)
```

Comment:
still to be fixed ...

Reply:
I don't see this timestamp as a problem. At least now it is stated that it has been 'chosen' and that is not completely arbitrary. But shows that `orig_time` is under the hood this kind of ugly floats.

Reply:
Sorry to insist here. 

But if a user wanted to use the `Annotations` class, how do you propose that they would set `orig_time`? I don't think anyone could dream up this number. If the idea to expose this parameter is that allows people to adjust the `orig_time` if it's not aligned. You could perhaps make it `annot.orig_time - 0.2` to shift orig time ...